pkg/agent/multicast/mcast_controller_test.go

```diff
@@ -209,6 +280,154 @@ func TestCheckLastMember(t *testing.T) {
 	}
 }
 
+func TestGetGroupPods(t *testing.T) {
+	now := time.Now()
+
+	mctrl := newMockMulticastController(t, false)
+	err := mctrl.initialize(t)
+	assert.Nil(t, err)
+	for _, tc := range []struct {
+		groupMemberStatuses  []*GroupMemberStatus
+		interfaceNameMap     map[string]*interfacestore.InterfaceConfig
+		expectedGroupPodsMap map[string][]v1beta2.PodReference
+	}{
+		{
+			groupMemberStatuses: []*GroupMemberStatus{
+				{
+					group:        net.ParseIP("224.96.1.2"),
+					localMembers: map[string]time.Time{if1.InterfaceName: now.Add(-10 * time.Second), if2.InterfaceName: now.Add(-20 * time.Second)},
+				},
+				{
+					group:        net.ParseIP("224.96.1.3"),
+					localMembers: map[string]time.Time{if2.InterfaceName: now.Add(-20 * time.Second)},
+				},
+			},
+			interfaceNameMap: map[string]*interfacestore.InterfaceConfig{
+				if2.InterfaceName: if2,
+				if1.InterfaceName: if1,
+			},
+			expectedGroupPodsMap: map[string][]v1beta2.PodReference{
+				"224.96.1.2": {{Name: if2.PodName, Namespace: if2.PodNamespace}, {Name: if1.PodName, Namespace: if1.PodNamespace}},
+				"224.96.1.3": {{Name: if2.PodName, Namespace: if2.PodNamespace}},
+			},
+		},
+	} {
+		for _, g := range tc.groupMemberStatuses {
+			err := mctrl.groupCache.Add(g)
+			assert.Nil(t, err)
+		}
+		for k, v := range tc.interfaceNameMap {
+			mockIfaceStore.EXPECT().GetInterfaceByName(k).AnyTimes().Return(v, true)
+		}
+		groupPodsMap := mctrl.GetGroupPods()
+		assert.Equal(t, len(tc.expectedGroupPodsMap), len(groupPodsMap))
+		for k, v := range groupPodsMap {
+			assert.ElementsMatch(t, tc.expectedGroupPodsMap[k], v)
+		}
+	}
+}
+
+func TestGetPodStats(t *testing.T) {
+	mctrl := newMockMulticastController(t, false)
+	err := mctrl.initialize(t)
+	assert.Nil(t, err)
+
+	for _, tc := range []struct {
+		iface            *interfacestore.InterfaceConfig
+		egressPodStats   *types.RuleMetric
+		ingressPodStats  *types.RuleMetric
+		expectedPodStats *PodTrafficStats
+	}{
+		{
```

Comment:
If there is only one case, no need to use for-loop ?

Reply:
updated